applications/nrf5340_audio/src/audio/audio_system.c

```diff
@@ -33,7 +34,8 @@ LOG_MODULE_REGISTER(audio_system, CONFIG_AUDIO_SYSTEM_LOG_LEVEL);
 K_THREAD_STACK_DEFINE(encoder_thread_stack, CONFIG_ENCODER_STACK_SIZE);
 
 DATA_FIFO_DEFINE(fifo_tx, FIFO_TX_BLOCK_COUNT, WB_UP(BLOCK_SIZE_BYTES));
-DATA_FIFO_DEFINE(fifo_rx, FIFO_RX_BLOCK_COUNT, WB_UP(BLOCK_SIZE_BYTES));
+DATA_FIFO_DEFINE(fifo_rx, FIFO_RX_BLOCK_COUNT, WB_UP(sizeof(struct audio_data)));
+NET_BUF_POOL_FIXED_DEFINE(fifo_rx_pool, FIFO_RX_BLOCK_COUNT, FRAME_SIZE_BYTES, 0, NULL);
```

Comment:
Did you consider the approach of having the data storage in the `audio_data` struct, instead of separating the data from the struct and using the pointer?

Reply:
That would of course require significant more stack memory if doing things like `struct audio_data audio_frame = {0};`